src/main/java/org/graylog/integrations/dataadpaters/GreyNoiseDataAdapter.java

```diff
@@ -0,0 +1,202 @@
+import com.unboundid.util.json.JSONException;
+import com.unboundid.util.json.JSONObject;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import org.graylog2.plugin.lookup.LookupCachePurge;
+import org.graylog2.plugin.lookup.LookupDataAdapter;
+import org.graylog2.plugin.lookup.LookupDataAdapterConfiguration;
+import org.graylog2.plugin.lookup.LookupResult;
+import org.graylog2.security.encryption.EncryptedValue;
+import org.graylog2.security.encryption.EncryptedValueService;
+import org.joda.time.Duration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.validation.constraints.NotEmpty;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+public class GreyNoiseDataAdapter extends LookupDataAdapter {
+    private static final Logger LOG = LoggerFactory.getLogger(GreyNoiseDataAdapter.class);
+    public static final String NAME = "GreyNoise";
+    static final String GREYNOISE_IPQC_ENDPOINT = "https://api.greynoise.io/v2/noise/quick/";
+
+    private final EncryptedValueService encryptedValueService;
+    private final Config config;
+    private OkHttpClient okHttpClient;
+
+    @Inject
+    public GreyNoiseDataAdapter(@Assisted("id") String id,
+                                @Assisted("name") String name,
+                                @Assisted LookupDataAdapterConfiguration config,
+                                MetricRegistry metricRegistry,
+                                EncryptedValueService encryptedValueService) {
+        super(id, name, config, metricRegistry);
+        this.config = (Config) config;
+        this.encryptedValueService = encryptedValueService;
+    }
+
+    @Override
+    public void doStart() throws Exception {
+
+        okHttpClient = new OkHttpClient().newBuilder()
+                                         .build();
+
+    }
+
+    @Override
+    public void doStop() throws Exception {
+
+    }
+
+    @Override
+    public Duration refreshInterval() {
+        return Duration.ZERO;
+    }
+
+    @Override
+    protected void doRefresh(LookupCachePurge cachePurge) throws Exception {
+
+    }
+
+    @Override
+    protected LookupResult doGet(Object keyObject) {
+        try {
+
+            Request request = new Request.Builder()
+                    .url(GREYNOISE_IPQC_ENDPOINT + keyObject.toString())
+                    .method("GET", null)
+                    .addHeader("Accept", "application/json")
+                    .addHeader("key", encryptedValueService.decrypt(config.apiToken()))
+                    .build();
+
+            return parseResponse(okHttpClient.newCall(request).execute());
+
+        } catch (Exception e) {
+            LOG.error("An error occurred while retrieving lookup result [{}]", e.toString());
+            return LookupResult.withError();
+        }
```

Comment:
It's generally a good practice to ensure the `Response` object returned from `.execute()` gets closed.  This is a good place to use Java's try-with-resources functionality:

```suggestion
        Request request = new Request.Builder()
                .url(GREYNOISE_IPQC_ENDPOINT + keyObject.toString())
                .method("GET", null)
                .addHeader("Accept", "application/json")
                .addHeader("key", encryptedValueService.decrypt(config.apiToken()))
                .build();

        try (Response response = okHttpClient.newCall(request).execute()) {

            return parseResponse(response);

        } catch (Exception e) {
            LOG.error("An error occurred while retrieving lookup result [{}]", e.toString());
            return LookupResult.withError();
        }
```

Declaring the `Response` object in `try (...) {` will ensure it automatically gets closed when the try/catch block finishes.